api/src/main/java/org/apache/iceberg/expressions/ExpressionVisitors.java

```diff
@@ -89,11 +91,11 @@ public R or(R leftResult, R rightResult) {
       return null;
     }
 
-    public <T> R in(BoundReference<T> ref, Literal<T> lit) {
+    public <T> R in(BoundReference<T> ref, Set<Literal<T>> lits) {
       return null;
```

Comment:
I think this should throw `UnsupportedOperationException` instead of returning null by default. We don't want a visitor that was written before `in` and `notIn` were supported to behave strangely because of unexpected null values returned by these default implementations.

Reply:
Yeah, I agree. 
Although I went through all concrete classes and fixed them (as I changed the signature), it is better to throw exception instead of returning `null`.
It seems better to throw `UnsupportedOperationException` for default implementations for all operations. I can make those changes if there is no specific reason to `return null`.
Let me know your comments. Thanks.

Reply:
Let's make that change separately, not in this commit.

Reply:
👌